ChatSecure/Classes/View Controllers/OTRMessagesViewController.m

```diff
@@ -1945,4 +1929,37 @@ - (IBAction)didPressMigratedSwitch {
     }
 }
 
+// If we find any incoming migration link messages, show the "your friend has migrated" header
+// to allow the user to start chatting with the new account instead.
+- (void) checkRangeForMigrationMessage:(NSRange)range {
```

Comment:
This can still cause stuttering because of the database fetch `messageAtIndexPath:` operates serially. I'd suggest passing in a messageText string parameter. Also, I don't think the ivar for linkDetector is needed, you could just allocate a temporary one in here, and I'm not sure if its thread safe to reuse it anyway.

Is this possible to move this change downstream as well into ZomMessagesViewController?